toolchain/sem_ir/inst.h

```diff
@@ -445,6 +453,29 @@ class InstBlockStore : public BlockValueStore<InstBlockId> {
   }
 };
 
+inline auto CarbonHashValue(const Inst& value, uint64_t seed) -> HashCode {
+  Hasher hasher(seed);
+  hasher.Hash(value);
+  return static_cast<HashCode>(hasher);
+}
+
 }  // namespace Carbon::SemIR
 
+template <>
+struct llvm::DenseMapInfo<Carbon::SemIR::Inst> {
+  using Inst = Carbon::SemIR::Inst;
+  static auto getEmptyKey() -> Inst {
+    return Inst(-1, Carbon::SemIR::TypeId::Invalid, 0, 0);
+  }
+  static auto getTombstoneKey() -> Inst {
+    return Inst(-2, Carbon::SemIR::TypeId::Invalid, 0, 0);
+  }
```

Comment:
Instead of putting custom int values into the InstKind, had you considered something like a new InstKind::Invalid and instead modifying arg0?

Alternately, should this be validating that we never allow these values as an InstKind?

Reply:
Added a CHECK to the `Inst` constructor.